File: learner-activity-challenge/leraner-activity-js/challenge-solution.js
function evaluateLearnerActivity(activityArray) {
  // Return an array containing 3 values:
  // 1. number of days that the user completed a lesson
  // 2. longest streak
  // 3. level based on the last 7

  //   User status rules
  // <= 2 'Slow starter' [0,1,2]
  // >2 && <= 4 'Getting there' [3, 4]
  // >4 'Super user' [5,6 7]

  // Instructions
  // run npm install to ensure that you install Jest, the test framework
  // if you want to test you code, run 'npm test' from the console and look at the console output to see if your tests passed

  //   Example input
  //  [1,0,1,0,0,0,1,1,0,0,1,1,1,0,0,1,1,1,1,1,0,1,0]

  // Example output
  // [13, 5, 'Super user']

  let activitySummary = [];

  let daysActive = getNumberOfDaysActive(activityArray);
  let longestStreak = getLongestStreak(activityArray);
  let status = getUserStatus(activityArray);

  activitySummary.push(daysActive);
  activitySummary.push(longestStreak);
  activitySummary.push(status);

  return activitySummary;
}

function getNumberOfDaysActive(activityArray) {
  activityArray = activityArray.filter(x => {
    return x === 1;
  });
  return activityArray.length;
}

function getLongestStreak(activityArray) {
  let streak = 0;
  let longestStreak = 0;
  activityArray.forEach(element => {
    streak = element === 1 ? ++streak : 0;
    longestStreak = streak > longestStreak ? streak : longestStreak;
  });
  return longestStreak;
}

function getUserStatus(activityArray) {
  let lastSevenDays = activityArray.slice(activityArray.length - 7);
  lastSevenDays = lastSevenDays.filter(x => {
    return x === 1;
  });

  let status =
    lastSevenDays.length <= 2
      ? "Slow starter"
      : lastSevenDays.length > 2 && lastSevenDays.length <= 4
      ? "Getting there"
      : "Super user";

  return status;
}

// do not remove or change name
module.exports = evaluateLearnerActivity;

File: learner-activity-challenge/learner-activity-ruby/challenge_solution.rb

class LearnerHelper
    @@learner_status = {:slow_starter => 'Slow starter', :getting_there => 'Getting there', :super_user => 'Super user'}

    def self.evaluate_learner_activity(activity_summary)
        learner_activity = []
        learner_activity.push(get_number_days_active(activity_summary))
        learner_activity.push(get_longest_streak(activity_summary))
        learner_activity.push(get_user_status(activity_summary))
    end

    def self.get_number_days_active(activity_summary)
        days_active = activity_summary.select do |element|
            element == 1
        end
        days_active.length()
    end

    def self.get_longest_streak(activity_summary)
        streak = 0
        longest_streak = 0
 
        activity_summary.each do |element| 
            streak = element == 1 ? (streak += 1) : 0
            longest_streak = streak > longest_streak ? streak : longest_streak
        end
        longest_streak
    end

    def self.get_user_status(activity_summary) 
        status = ''
        last_seven_days = activity_summary.slice((activity_summary.length()-7), activity_summary.length())
        # print last_seven_days
        last_seven_days = last_seven_days.select do |x|
            x == 1 
        end 
        if last_seven_days.length() <= 2
            status = @@learner_status[:slow_starter]
        elsif last_seven_days.length() > 2 && last_seven_days.length() <= 4
            status = @@learner_status[:getting_there]
        else 
            status = @@learner_status[:super_user]
        end
        status 
    end
end
  